cli/loader_lib.c

```diff
@@ -349,10 +349,16 @@ static char *libstdcxxprobe(void)
             pid_t npid = waitpid(pid, &wstatus, 0);
             if (npid == -1) {
                 if (errno == EINTR) continue;
-                if (errno != EINTR) {
-                    perror("Error during libstdcxxprobe in parent process:\nwaitpid");
-                    exit(1);
+                if (errno == ECHILD) {
+                    // SIGCHLD is set to SIG_IGN or has flag SA_NOCLDWAIT, so the child
```

Comment:
This can be worked around by e.g. using clone on linux and the raw process APIs on osx (and possibly freebsd) if we want to.

Reply:
I find this `fork()` (or clone, etc.) fairly suspect to begin with TBH

As we try to be embeddable / loadable in larger applications it becomes more likely that (1) `libstdc++` has already been loaded and we're too late to do anything, and (2) the address space of the application is large and mutating, so that `fork()` can become potentially expensive

I'd prefer to try to remove our dependency on `libstdc++` for small, simple libraries TBH and get rid of this altogether

Reply:
Even better, we should probably just `mmap` the `.so` and check its symbol table, rather than trying to do this sandboxed `dlopen`.

The hard part is emulating the linker search path though..